core/templates/pages/exploration-editor-page/feedback-tab/services/thread-data.service.ts

```diff
@@ -44,232 +49,279 @@ interface SuggestionAndFeedbackThreads {
+      // TODO(#8016): Move this this.http call
+      // to a backend-api.service with unit
+      // tests.
+      let suggestionsPromise = this.http.get(
+        this.getSuggestionListHandlerUrl(), {
           params: {
             target_type: 'exploration',
-            target_id: ContextService.getExplorationId()
+            target_id: this.contextService.getExplorationId()
           }
-        });
-        // TODO(#8016): Move this $http call to a backend-api.service with unit
-        // tests.
-        let threadsPromise = $http.get(getThreadListHandlerUrl());
+        }).toPromise();
+      // TODO(#8016): Move this
+      // this.http call to a backend-api.service with unit
+      // tests.
+      let threadsPromise = this.http.get(this.getThreadListHandlerUrl()).toPromise();
 
-        return $q.all([suggestionsPromise, threadsPromise]).then(response => {
+      return Promise.all([suggestionsPromise, threadsPromise])
+        .then(response => {
           let [suggestionData, threadData] = response.map(r => r.data);
           let suggestionBackendDicts: SuggestionBackendDict[] = (
             suggestionData.suggestions);
           let feedbackThreadBackendDicts = threadData.feedback_thread_dicts;
-          let suggestionThreadBackendDicts = threadData.suggestion_thread_dicts;
+          let suggestionThreadBackendDicts =
+          threadData.suggestion_thread_dicts;
 
           let suggestionBackendDictsByThreadId = new Map(
             suggestionBackendDicts.map(dict => [
-              SuggestionsService.getThreadIdFromSuggestionBackendDict(dict),
+              this.suggestionsService.getThreadIdFromSuggestionBackendDict(
+                dict),
               dict
             ]));
 
           return {
             feedbackThreads: feedbackThreadBackendDicts.map(
-              dict => setFeedbackThreadFromBackendDict(dict)),
+              dict => this.setFeedbackThreadFromBackendDict(dict)),
             suggestionThreads: suggestionThreadBackendDicts.map(
-              dict => setSuggestionThreadFromBackendDicts(
+              dict => this.setSuggestionThreadFromBackendDicts(
                 dict, suggestionBackendDictsByThreadId.get(dict.thread_id)))
           };
-        },
-        () => $q.reject('Error on retrieving feedback threads.'));
-      },
+        }, (err) => {
+          throw ('Error on retrieving feedback threads.');
+        });
+    }
 
-      getMessagesAsync: function(thread: AnyThread): Promise<ThreadMessage[]> {
-        if (!thread) {
-          throw new Error('Trying to update a non-existent thread');
-        }
-        let threadId = thread.threadId;
-        // TODO(#8016): Move this $http call to a backend-api.service with unit
-        // tests.
-        return $http.get(getThreadHandlerUrl(threadId)).then(response => {
-          let threadMessageBackendDicts = response.data.messages;
+    errorCallback(error):void{
```

Comment:
what is this for?

Reply:
the errorCallback is a function used to throw err whenever there is need to, it is used in other function, like this.errorCallback(rrorResponse.error.error) which is used in the function directly below the errorCallback function